mart/callbacks/freeze.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from __future__ import annotations
+
+import torch
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from mart import utils
+
+logger = utils.get_pylogger(__name__)
+
+__all__ = ["FreezeModule"]
+
+
+class FreezeModule(Callback):
+    def __init__(
+        self,
+        module="backbone",
+    ):
+        self.name = module
+
+    def setup(self, trainer, pl_module, stage):
+        # FIXME: Use DotDict?
+        module = getattr(pl_module.model, self.name, None)
+
+        if module is None or not isinstance(module, torch.nn.Module):
+            raise MisconfigurationException(
+                f"The LightningModule should have a nn.Module `{self.name}` attribute"
+            )
+
+        for name, param in module.named_parameters():
+            logger.debug(f"Disabling gradient for {name}")
+            param.requires_grad_(False)
+
+        for name, module in module.named_modules():
+            module_kind = module.__class__.__name__
+            if "BatchNorm" in module_kind:
```

Comment:
We should probably make this configurable. For example, YoloV4 uses DropBlock: https://github.com/AlexeyAB/Yet-Another-YOLOv4-Pytorch/blob/d80d6a20372598b6306b37218cb61533e8bd9592/model.py#L102.